### PR TITLE
Rename variables in diff-report.sh to support new workflow

### LINKED ISSUE #15261: Refactor diff-report.sh to support new workflow
Refactor `DIFF_PROJECTS_LINK `to `LINKS_FROM_PR`

### PR DESCRIPTION
**Pull request by relentless-pursuit**
Resolves #15261

### CODE REVIEW

**Comment by relentless-pursuit** [comment]
@romani refactoring won't help. 

https://github.com/checkstyle-gsoc/checkstyle/actions/runs/9927719539/job/27423024630

We need to enable the condition of a backwards compatibility. I tried doing just the refactoring and tried running the workflow and it failed. Attached the link above. If you want, i can make the changes earlier done as a part of PR of workflow here.

File: .ci/diff-report.sh [hunk 1]
```diff
@@ -40,7 +40,7 @@ download-files)
   echo "Downloading files..."
 
   # check for projects link from PR, if not found use default from contribution repo
-  LINK="${LINK_FROM_PR:-$DEFAULT_PROJECTS_LINK}"
+  LINK="${DIFF_PROJECTS_LINK:-$DEFAULT_PROJECTS_LINK}"
```
Comment (romani) [inline .ci/diff-report.sh:43]: you need to change action code too, LINK_FROM_PR should be completely replaced with new value DIFF_PROJECTS_LINK

https://github.com/checkstyle/checkstyle/pull/15235#discussion_r1676976750

**Review by romani — CHANGES_REQUESTED** [review]
items:

**Comment by romani** [comment]
>We need to enable the condition of a backwards compatibility

please look at your README, 
https://github.com/checkstyle/test-configs/tree/main/AbbreviationAsWordInName/Example1

there are two links (two variables) need to be defined to let diff-report.sh do a job. So new workflow will just set such two variables in slightly different way.

old name was   confusing as it was mentioned not what is inside but from where it was taken, it was mistake. As you rename it, sh script will become agnostic on original location of links.

**Comment by relentless-pursuit** [comment]
> LINK="${DIFF_PROJECTS_LINK:-$DEFAULT_PROJECTS_LINK}"

LINK="${DIFF_PROJECTS_LINK:-$DEFAULT_PROJECTS_LINK}"


he new workflow doesn't require this DEFAULT_PROJECT_LINK. Shall i remove it?
  `LINK="${DIFF_PROJECTS_LINK:-$DEFAULT_PROJECTS_LINK}"`

**Comment by romani** [comment]
>Shall i remove it?

no, we do not change shell file more that renaming.

**Comment by relentless-pursuit** [comment]
> > Shall i remove it?
> 
> no, we do not change shell file more that renaming. it will not harm to keep it as you will always set it by your action, in most worse case it will fallback to default that is better than nothing

But, upon reproducing the same in my own repo, this fails.

**Review by romani — APPROVED** [review]
ok to merge.

there were only two places for this name https://github.com/search?q=repo%3Acheckstyle%2Fcheckstyle%20LINK_FROM_PR&type=code

image

**Comment by romani** [comment]
> But, upon reproducing the same in my own repo, this fails.

links to code, PR with comment  and execution in github action

**Comment by relentless-pursuit** [comment]
> > But, upon reproducing the same in my own repo, this fails.
> 
> links to code, PR with comment and execution in github action

https://github.com/checkstyle-gsoc/checkstyle/blob/master/.github/workflows/Report-By-Comment.yml
https://github.com/checkstyle-gsoc/checkstyle/blob/master/.ci/diff-report.sh

workflow runner:
https://github.com/checkstyle-gsoc/checkstyle/actions/runs/9927719539/job/27423024630

**Comment by romani** [comment]
lets move your debug issues to https://github.com/checkstyle-gsoc/checkstyle/pull/1#issuecomment-2227381776

**Pull request merged**